### PR TITLE
[RHELC-1061, RHELC-1060, RHELC-1062] Allow package names with periods

### LINKED ISSUE #854: FYI, converting from Rocky 8, issue with packages that have periods in the name
https://github.com/oamg/convert2rhel/blob/7fba2cbaa2d62d70898059954b36a386fe42657f/convert2rhel/pkgmanager/handlers/dnf/__init__.py#L123

We're seeing the conversion barf on packages like:

python3.11-pyyaml.x86_64
python3.11-libs.x86_64
java-1.8.0-openjdk.x86_64

etc. 

Evidently the period in the package name is messing up the python tuple(pkg.rsplit(".")) since the split gives 3 args instead of the two.

We worked around it by uninstalling the python3.11 and java-1.8.0 various packages...

### PR DESCRIPTION
**Pull request by nikatbu**
Package names with periods (ex: python3.11) cause issues with the pkg.rsplit(".") as it splits into 3 or more instead of just the 2 package and name.

Example error:

```
(ERROR) VALIDATE_PACKAGE_MANAGER_TRANSACTION.UNEXPECTED_ERROR: Unhandled exception was caught: too many values to unpack (expected 2)
Please file a bug at https://issues.redhat.com/ to have this fixed or
    a specific error message added.
Traceback: Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/convert2rhel/actions/__init__.py", line 393, in run  action.run()
  File "/usr/lib/python3.6/site-packages/convert2rhel/actions/pre_ponr_changes/transaction.py", line 46, in run  validate_transaction=True,
  File "/usr/lib/python3.6/site-packages/convert2rhel/pkgmanager/handlers/dnf/__init__.py", line 210, in run_transaction  self._perform_operations()
  File "/usr/lib/python3.6/site-packages/convert2rhel/pkgmanager/handlers/dnf/__init__.py", line 123, in _perform_operations name, arch = tuple(pkg.rsplit("."))
ValueError: too many values to unpack (expected 2)
```

Fix is to do rsplit("."), 1 to split into name, arch where name could still contain periods.

Jira Issues: 
- [RHELC-1060](https://issues.redhat.com/browse/RHELC-1060)
- [RHELC-1061](https://issues.redhat.com/browse/RHELC-1061)
- [RHELC-1062](https://issues.redhat.com/browse/RHELC-1062)

Fixes #854 

Checklist

- [X] PR has been tested manually in a VM (either author or reviewer)
- [ ] Jira issue has been made public if possible
- [X] `[RHELC-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [ ] PR title explains the change from the user's point of view
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

**Comment by nikatbu** [comment]
Fixes #854

File: convert2rhel/pkgmanager/handlers/dnf/__init__.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def _perform_operations(self):
         for pkg in original_os_pkgs:
             # Splitting the name and arch so we can filter it out in the list
             # of packages to upgrade.
-            name, arch = tuple(pkg.rsplit("."))
+            name, arch = tuple(pkg.rsplit("."), 1)
```
Comment (SpyTec) [inline convert2rhel/pkgmanager/handlers/dnf/__init__.py:123]: This should be
```suggestion
            name, arch = tuple(pkg.rsplit(".", 1))
```

**Review by SpyTec — CHANGES_REQUESTED** [review]
Thanks for fixing it! Got one small thing left

We will keep the PR open so we can add tests to this PR to prevent regression, hope that's fine with you!

File: convert2rhel/pkgmanager/handlers/dnf/__init__.py (hunk 1 above)
Comment (nikatbu) [reply convert2rhel/pkgmanager/handlers/dnf/__init__.py:123]: Thanks for the fix. Sorry for the typo, I had it working locally then messed up when I translated it to the PR.  Oops. Good catch, thanks.

**Comment by nikatbu** [comment]
Hi, let me know if I'm supposed to put a label on this as "bug-fix". I couldn't figure out how to do that, it almost seems like this is restricted to only the owners/maintainers of the repo.

**Comment by r0x0d** [comment]
As an output from a conversation in our internal chat, we need to update the unit_tests/integration tests  to have the following names to be parsed: 
```
python3.11-3.11.2-2.el8.x86_64
python3.11-six-1.16.0-1.el8.noarch
python3.11-cffi-1.15.1-1.el8.x86_64
python3.11-libs-3.11.2-2.el8.x86_64
python3.11-ply-3.11-1.el8.noarch
python3.11-setuptools-wheel-65.5.1-2.el8.noarch
python3.11-pip-wheel-22.3.1-2.el8.noarch
python3.11-cryptography-37.0.2-5.el8.x86_64
java-1.8.0-openjdk-headless-1.8.0.372.b07-4.el8.x86_64
python3.11-pyyaml-6.0-1.el8.x86_64
python3.11-pycparser-2.20-1.el8.noarch
```

I will see if I can manage to push to this PR by the end of the week with the updates to the necessary tests

**Comment by r0x0d** [comment]
> Hi, let me know if I'm supposed to put a label on this as "bug-fix". I couldn't figure out how to do that, it almost seems like this is restricted to only the owners/maintainers of the repo.

Done! Thanks for commenting it out.

**Review by SpyTec — APPROVED** [review]

**Pull request merged**